distributed/active_memory_manager.py

```diff
@@ -44,7 +47,7 @@ class ActiveMemoryManagerExtension:
     # Current memory (in bytes) allocated on each worker, plus/minus pending actions
     workers_memory: dict[WorkerState, int]
     # Pending replications and deletions for each task
-    pending: defaultdict[TaskState, tuple[set[WorkerState], set[WorkerState]]]
+    pending: dict[TaskState, tuple[set[WorkerState], set[WorkerState]]]
```

Comment:
This is mostly a style/robustness change. The defaultdict was causing more annoyance than it was solving.